presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -408,6 +408,11 @@ default ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connect
         throw new PrestoException(NOT_SUPPORTED, "This connector does not support inserts");
     }
 
+    default ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle, List<ColumnHandle> columns)
```

Comment:
Shouldn't you deprecate the other method?

Reply:
The change is that simple that I think you could remove the other method.

Reply:
That would be a SPI breaking change.